app/src/main/java/org/fossasia/susi/ai/Login/LoginActivity.kt

```diff
@@ -0,0 +1,165 @@
+package org.fossasia.susi.ai.Login
+
+import android.app.ProgressDialog
+import android.content.DialogInterface
+import android.content.Intent
+import android.graphics.Color
+import android.os.Bundle
+import android.support.design.widget.TextInputLayout
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.AutoCompleteTextView
+import android.widget.Button
+import android.widget.RadioButton
+import butterknife.BindView
+import butterknife.ButterKnife
+import butterknife.OnClick
+import butterknife.OnEditorAction
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.activities.ForgotPasswordActivity
+import org.fossasia.susi.ai.activities.MainActivity
+import org.fossasia.susi.ai.activities.SignUpActivity
+import org.fossasia.susi.ai.helper.Constant
+
+/**
+ * Created by chiragw15 on 4/7/17.
+ */
+class LoginActivity : AppCompatActivity(), ILoginView {
+
+    @BindView(R.id.email)
+    lateinit var email: TextInputLayout
+    @BindView(R.id.email_input)
+    lateinit var autoCompleteEmail: AutoCompleteTextView
+    @BindView(R.id.password)
+    lateinit var password: TextInputLayout
+    @BindView(R.id.log_in)
+    lateinit var logIn: Button
+    @BindView(R.id.susi_default)
+    lateinit var susiServer: RadioButton
+    @BindView(R.id.personal_server)
+    lateinit var personalServer: RadioButton
+    @BindView(R.id.input_url)
+    lateinit var url: TextInputLayout
+
+    var loginPresenter: ILoginPresenter? = null
+    var progressDialog: ProgressDialog? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        ButterKnife.bind(this)
+
+        loginPresenter = LoginPresenter()
+        loginPresenter!!.onAttach(this)
```

Comment:
Don't use, !! Operator. It tries to invoke function even if value is null. Use '?' elvis operator instead, it will invoke method only if value is not null.